app/views/books/index.html.erb

```diff
@@ -15,8 +15,11 @@
 <ul class="books unavailable">
   <% @unavailable_books.each do |book| %>
     <li>
-      <%= book.title %>, 
+      <%= book.title %>,
       <%= book.author %>
+      <% if current_user.checkouts.where(book: book).present? %>
```

Comment:
There's a couple things worth discussing here.

1. We typically want to avoid `where` and the like in views. They are far removed from the model. Using `where` means the view is now coupled to the ActiveRecord implementation of `Checkout`.

2. This introduces an N+1 query. That is, `@unavailable_books` performs one query to get a list of books. Then for the `N` books we have we're going to end up doing `N` SQL queries to check for the existence of a checkout of that book by this user. When there's a small number of books, this is inefficient, but perhaps not noticeable. If there are 10, 100, 1000 books then the performance hit of the database round trip becomes a problem.

The quick way to fix both of these and maintain this UI is to do something like: `@current_user_book_ids = current_user.checkouts.pluck(:book_id)` in the controller. This will do a single query to return an array of book ids the user has a checkout for. Then here in the view do something like `if @current_user_book_ids.include?(book_id)`

The consultant in me would probably ask, "Is what we're doing here really the best way for a user to to return their books?" I'd probably expect to have a page dedicated to viewing the books *I* have checked out and the ability to return them? What's here might be the quickest way to get something workable, but I suspect I'd be technically happier with the dedicated view if it were a possibility.